### PR TITLE
use local codegen for protobuf message types

### LINKED ISSUE #1483: Bundle @buf interfaces into Protobuf
It is hard to track the `@buf` package versions. If not the IDE import substitutions, I would never learn the path from which I should import common things like `Metadata`, `Address`, `BalancesResponse`, etc.

It would be way easier to have only one entry point for such interfaces and `@penumbra-zone/protobuf` package is the great fit for it. We could improve two things at once:
1. Bundle the specific version of the `@buf/...` into a specific version of `@penumbra-zone/protobuf`. There won't be a need to separately install `@buf` packages by consumers and no more need to configure the `@buf` registry
2. Make the imports straightforward and only from one place

I don't have the exact solution on how to structure the `protobuf` package. One option is to put the interfaces next to the services it is used within. For example, `Metadata` is used in the `ViewService`, so it could be placed in `@penumbra-zone/protobuf/view-service`

### PR DESCRIPTION
**Pull request by turbocrime**
fixes #1483 #1503

replaces #1582 #1583

it's simplest to use `buf generate` to create the definitions

### CODE REVIEW

File: package.json
Comment (turbocrime) [inline package.json:49]: this one @buf dep is still necessary because the registry incorrectly fails to declare a dependency

File: packages/protobuf/package.json
Comment (turbocrime) [inline packages/protobuf/package.json:17]: versions used in core

https://github.com/penumbra-zone/penumbra/blob/main/proto/buf.lock

File: packages/protobuf/src/gen/.npmignore
Comment (turbocrime) [inline packages/protobuf/src/gen/.npmignore:1]: empty npmignore will allow this dir to be packaged

File: packages/protobuf/package.json
Comment (VanishMax) [inline packages/protobuf/package.json:21]: praise: thanks for this – will way easier for the rest of us to apply new bufs

File: packages/perspective/src/translators/swap-claim-view.ts
Comment (VanishMax) [inline packages/perspective/src/translators/swap-claim-view.ts:5]: question: any chance in the future we could simplify the imports? Maybe after applying v2 of the generator we could export types as barrels?

**Review by VanishMax — APPROVED** [review]
Great improvement!

This is ultimately a better way than how it was done in #1582. Was there a specific reason to not continue working or commenting on #1582 ?

**Comment by turbocrime** [comment]
going to wait until recent core message type changes are merged for ibc compat before trying to merge this

File: packages/perspective/src/translators/swap-claim-view.ts
Comment (turbocrime) [reply packages/perspective/src/translators/swap-claim-view.ts:5]: the service definitions have a barrel export defined, but i think trying to collect other types into barrels is likely to be very complex and not very useful.

File: apps/minifront/src/state/helpers.ts
Comment (turbocrime) [inline apps/minifront/src/state/helpers.ts:161]: this caused problems at some point during this work, and the types were technically incorrect, so it's now simplified

**Pull request merged**
